api/apps/api/src/modules/scenarios/scenarios.service.ts

```diff
@@ -86,7 +86,9 @@ export class ScenariosService {
 
   async create(input: CreateScenarioDTO, info: AppInfoDTO) {
     const validatedMetadata = this.getPayloadWithValidatedMetadata(input);
-    return this.crudService.create(validatedMetadata, info);
+    const scenario = await this.crudService.create(validatedMetadata, info);
+    await this.planningUnitsLinkerService.link(scenario);
```

Comment:
1 Isn't it necessary for update as well?

2 Seems like creating scenario has some "side" effects (or is just a part of creation process?). Would recommend putting it into standalone service to keep things together.

Reply:
on update of scenario, we shouldn't relink PUs to scenario, but it's true that we don't yet enforce that a scenario can only be created if the parent project has enough (and coherent) settings to let us attach a PU grid to the scenario. In practice this shouldn't be an issue with the current frontend flow so for the sake of getting closer to preparing demo projects and test running them in an actual instance I'd be keen to skip this for the moment.

I actually had a check for this in an earlier commit in this PR but I've dropped it now because it was not in the most appropriate place and it would break tests and I just can't update more tests and fixtures 🤯 (again, to get closer to run this whole workflow)

as for side effects, yes and no - I'd say these are side effects, but they are part of the scenario creation process. if the settings of the parent project that let us link PUs to a scenario change, we'll have (in a future iteration) to delete all the scenarios of the project that would be affected and cascade deletion donwstream.

Reply:
similar to another comment related to keeping the create/update logic in a single place: right now it is scattered around 3 services, among those one is solely `crud.service` but it doesn't fully create a scenario (the facade which calls it has the knowledge that some other services should be called)